service/resource/dnsrecord/conv.go

```diff
@@ -0,0 +1,33 @@
+package dnsrecord
+
+import (
+	"github.com/giantswarm/azuretpr"
+	"github.com/giantswarm/microerror"
+)
+
+func toCustomObject(v interface{}) (azuretpr.CustomObject, error) {
```

Comment:
IMO this should be inside the key package. At least the other operators are doing it this way. Also note that we should version the key packages as well then. Also, what does `conv.go` stand for? Would be nice to keep this with other operators aligned.

Reply:
It was copy/paste from other resource. I only moved that from `resource.go` to `conv.go`. conv stands for conversion. Maybe it isn't the best name since in Go terms it's type assertion.

I can move toCustomObject to `key` package, that makes sense, since it's reused between resources, but for `toDNSRecords` it's impossible since it returns a private type.

Reply:
`toDNSRecords` is about the resource so this is fine here. The custom object assertion should move to the key package though.